spec/inferno/web/serializers/test_group_spec.rb

```diff
@@ -17,7 +17,7 @@
     expect(serialized_group['description']).to eq(group.description)
     expect(serialized_group['short_description']).to eq(group.short_description)
     expect(serialized_group['input_instructions']).to eq(group.input_instructions)
-    expect(serialized_group['inputs'].length).to eq(group.inputs.length)
+    expect(serialized_group['inputs'].length).to eq(group.available_inputs.length)
```

Comment:
Thanks for pointing out `input_definitions`. It shouldn't exist anymore, so let's ignore it. Let's consider the following simple group:
```
group do
  input :a

  test do
    input :b
    output :c
  end

  test do
    input :c
  end
end
```
- `inputs` is a list of identifiers for the current runnable's inputs. It is used to create `attr_readers` on tests so that the inputs are easily available within a test's `run` block. It will include the inputs of any of its parents:
```
pry(main)> group.inputs
=> [:a]
pry(main)> group.tests.map(&:inputs)
=> [[:a, :b], [:a, :c]]
```
- `available_inputs` is a Hash of `identifier => Input` that contains all of the inputs that are available for a user to provide when running a particular runnable. It recursively looks through all of a runnable's children to determine all of their inputs, merges them together, and gets rid of any inputs that are supplied by a previous output (and therefore are not really available for the user to provide input for).
```
pry(main)> group.available_inputs
=> {
     :a=>#<Inferno::Entities::Input:0x00007fd2bc15c868 @name="a", @type="text">, 
     :b=>#<Inferno::Entities::Input:0x00007fd2bc167830 @name="b", @type="text">
   }
pry(main)> group.tests.first.available_inputs
=> {
     :a=>#<Inferno::Entities::Input:0x00007fd2bc167d58 @name="a", @type="text">, 
     :b=>#<Inferno::Entities::Input:0x00007fd2bc167830 @name="b", @type="text">
   }
pry(main)> group.tests.last.available_inputs
=> {
     :a=>#<Inferno::Entities::Input:0x00007fd2bc1669d0 @name="a", @type="text">, 
     :c=>#<Inferno::Entities::Input:0x00007fd2bc1664a8 @name="c", @type="text">
   }
```

Reply:
Great example, that definitely clears it up!